From 000034d7744e215b7336de13de1524487bc3462e Mon Sep 17 00:00:00 2001
From: "Andrew F. Davis" <afd@ti.com>
Date: Wed, 19 Feb 2020 15:32:01 -0500
Subject: [tiL5.10-P PATCH 1/2] dma-buf: heaps: Initialize during core instead
 of subsys

Some clients of DMA-Heaps probe earlier than subsys_initcall(), this
can cause issues when these clients call dma_heap_add() before the
core DMA-Heaps framework has initialized. DMA-Heaps should initialize
during core startup to get ahead of all users.

Signed-off-by: Andrew F. Davis <afd@ti.com>
Signed-off-by: Gowtham Tammana <g-tammana@ti.com>
---
 drivers/dma-buf/dma-heap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma-buf/dma-heap.c b/drivers/dma-buf/dma-heap.c
index afd22c9dbdcf..af6edfbeddfe 100644
--- a/drivers/dma-buf/dma-heap.c
+++ b/drivers/dma-buf/dma-heap.c
@@ -295,4 +295,4 @@ static int dma_heap_init(void)
 
 	return 0;
 }
-subsys_initcall(dma_heap_init);
+core_initcall(dma_heap_init);
-- 
2.32.0

